rai/node/node.hpp

```diff
@@ -23,7 +23,18 @@
 
 #include <miniupnpc.h>
 
-std::ostream & operator << (std::ostream &, std::chrono::system_clock::time_point const &);
+namespace rai
+{
+// time_point_wrapper - Used to provide an overload for `std::iostream`
+// for passing `std::chrono::system_clock::time_point`. Overloading
+// it directly would be undefined behavior as per [namespace.std]/1.
+struct time_point_wrapper
```

Comment:
would a `typedef` still be UB?

Reply:
Yes, typedefs are just alises, not user-defined types.